bed/compare.go

```diff
@@ -160,6 +161,27 @@ func CompareChromEndByChrom(a Bed, b Bed) int {
 	return CompareChromEnd(a, b)
 }
 
+//CompareDistance compares beds by chromStart and chromEnd and returns the minimum distance between
+//the two beds. return options; -1 (different chromosomes, no distance calculated),
+//0 (overlap, minimum distance is 0), >=0 is the minimum distance.
+func CompareDistance(a Bed, b Bed) int {
+	chromComp := strings.Compare(a.Chrom, b.Chrom)
+	if chromComp == 0 {
+		return -1
```

Comment:
I would give an error here to be more explicit that some unexpected happened, or that we are unable to do what the programmer asked.  I think you could either log.Fatal, or you could update your function to return an error along with an int.  Probably an error along with an int will be more useful in your case and others.